Neos.Flow/Documentation/TheDefinitiveGuide/PartIII/Routing.rst

```diff
@@ -729,10 +729,66 @@ two more options you can use:
 With ``suffix`` you can specify a custom filename suffix for the SubRoute. The ``variables`` option allows you to
 specify placeholders in the SubRoutes (see `Nested Subroutes`_).
 
+It also is possible to specify a `providerFactory` and (optional) `providerOptions` to generate the subroutes via the
+`Neos\Flow\Mvc\Routing\RoutesProviderFactoryInterface` and the ``Neos\Flow\Mvc\Routing\RoutesProviderInterface`.
+
+.. code-block:: yaml
+
+  Neos:
+    Flow:
+      mvc:
+        routes:
+          Vendor.Example:
+            position: 'before Neos.Neos'
+            providerFactory: \Neos\Flow\Mvc\Routing\AttributeRoutesProviderFactory
+            providerOptions:
+              classNames:
+                - Vendor\Example\Controller\ExampleController
+
 .. tip::
 
 	You can use the ``flow:routing:list`` command to list all routes which are currently active, see `CLI`_
 
+Subroutes from Annotations
+--------------------------
+
+The ``Flow\Route`` attribute allows to define routes directly on the affected method.
+
+.. code-block:: php
+
+  use Neos\Flow\Mvc\Controller\ActionController;
+  use Neos\Flow\Annotations as Flow;
+
+  class ExampleController extends ActionController
+  {
+      #[Flow\Route(uriPattern:'my/path', httpMethods: ['GET'])]
+      public function someAction(): void
+      {
+      }
+
+      #[Flow\Route(uriPattern:'my/other/b-path', defaults: ['test' => 'b'])]
+      #[Flow\Route(uriPattern:'my/other/c-path', defaults: ['test' => 'c'])]
+      public function otherAction(string $test): void
+      {
+      }
+  }
```

Comment:
ill try to add a functional test where once can see the feature 1 to 1 in action. Currently the unit tests are cool to have but really obscure :D